### PR TITLE
fix a bug for image input when computing logp

### PR DESCRIPTION
**Pull request by Yifan-Song793**
Modify `_get_per_token_logps` method in GRPOTrainer to fix a bug in computing logp. Also add `attention_mask` when computing logp, which may fix the bug for bsz>1 (need review)

### CODE REVIEW

**Comment by chenllliang** [comment]
both _get_per_token_logps looks good to me. After testing, the inital reward is still low(0.2) when bsz=2. Would continue to warn user to keep bsz=1 for now. @Yifan-Song793
